server/service/http/api.go

```diff
@@ -158,15 +160,36 @@ func (a *API) respondError(w http.ResponseWriter, apiErr coderr.CodeError, msg s
 	}
 }
 
+func (a *API) getLeaderAddr(writer http.ResponseWriter, req *http.Request) {
+	leaderAddr, err := a.forwardClient.GetLeaderAddr(req.Context())
+	if err != nil {
+		log.Error("get leader addr failed", zap.Error(err))
+		a.respondError(writer, member.ErrGetLeader, fmt.Sprintf("err: %s", err.Error()))
+		return
+	}
+	a.respond(writer, leaderAddr)
+}
+
 type GetShardTables struct {
 	ClusterName string   `json:"clusterName"`
-	NodeName    string   `json:"nodeName"`
 	ShardIDs    []uint32 `json:"shardIDs"`
 }
 
 func (a *API) getShardTables(writer http.ResponseWriter, req *http.Request) {
+	resp, isLeader, err := a.forwardClient.forwardToLeader(req)
+	if err != nil {
+		log.Error("forward to leader failed", zap.Error(err))
+		a.respondError(writer, ErrForwardToLeader, fmt.Sprintf("err: %s", err.Error()))
+		return
+	}
```

Comment:
Add a util function.

Reply:
I think the logic here is simple enough, and there is no need to encapsulate it. In the future, it may be considered to put the forward logic directly in it through the pre-processing of the HTTP framework.